### ISSUE TITLE
Consider AssemblyInfo.cs as a generated file

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#4184]
Based on [internal discussion](https://sonarsource.slack.com/archives/C012KBFFYD6/p1616671492054500), consider `AssemblyInfo` (CS and VB) as a generated file.

That should prevent it from being marked as duplicated code blocks, when AnalyzeGenerated=false.

Add Java integration tests to check duplications /  improve our [AutoGenerated ITs](https://github.com/SonarSource/sonar-dotnet/blob/1af2e34548c0873aa437a11d02f9d51373ed7177/its/src/test/java/com/sonar/it/csharp/AutoGeneratedTest.java#L34) to account for Copy Paste Detection metrics

**Comment by Corniel** [comment]
Shouldn't it also be considered a deprecated feature? (I know, that should be a different rule) It is only added for the pre-2017 project files. That in it self might be a constraint to: because if you have an AssemblyInfo file, and have a 2017-project file, it was added manually. And it might be worth checking if no class/struct/enum was defined, to mark it a such too.

**Comment by pavel-mikula-sonarsource** [comment]
I think there's nothing wrong with naming your file like that for the same purpose in new SDK-style projects. We actually have similar one https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.CSharp/Properties/AssemblyInfo.cs as there are few rules asking you to mark your assembly with assembly attributes (like S3904).

**Comment by Corniel** [comment]
@pavel-mikula-sonarsource I fully agree that you should be able to give your file the name you want (although there are guidelines about having the same name for your file as the class that defined in it, handled by StyleCop if I'm not mistaken).

My suggestion was (and I have to admit, not spelled out that well) that the assembly attributes previously required (and generated in the AssemblyInfo file) should become obsolete, and that deserves a rule on its own.

And for the auto generated part: you only want to skip those AssemblyInfo files, that where - as matter of fact - generated. If you decided to have you custom AssemblyInfo file, containing (only) some custom chosen attributes (like the one you mentioned), I think it's fair to not mark those as generated.

**Comment by AraHaan** [comment]
> Based on [internal discussion](https://sonarsource.slack.com/archives/C012KBFFYD6/p1616671492054500), consider `AssemblyInfo` (CS and VB) as a generated file.
> 
> That should prevent it from being marked as duplicated code blocks, when AnalyzeGenerated=false.
> 
> Add Java integration tests to check duplications / improve our [AutoGenerated ITs](https://github.com/SonarSource/sonar-dotnet/blob/1af2e34548c0873aa437a11d02f9d51373ed7177/its/src/test/java/com/sonar/it/csharp/AutoGeneratedTest.java#L34) to account for Copy Paste Detection metrics

So, those include files marked with ``// <autogenerated/>`` that come from source generators too?

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1917](https://sonarsource.atlassian.net/browse/NET-1917)

[NET-1917]: https://sonarsource.atlassian.net/browse/NET-1917?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
